src/Client.js

```diff
@@ -67,6 +67,19 @@ class Client extends EventEmitter {
     async initialize() {
         let [browser, page] = [null, null];
         
+        // eslint-disable-next-line no-useless-escape
+        const foldernameRegex = /^(?!.{256,})(?!(aux|clock\$|con|nul|prn|com[1-9]|lpt[1-9])(?:$|\.))[^ ][ \.\w-$()+=[\];#@~,&amp;']+[^\. ]$/i;
+
+        if (this.id && !foldernameRegex.test(this.id)) throw TypeError('Invalid client ID. Make sure you abide by the folder naming rules of your operating system.');
```

Comment:
I don't think a `TypeError` is suitable here, since even if the regex doesn't match it would still be the right type (`string`)

Reply:
hmm then just a "Error"?

Reply:
resolved